ws/test_ws_ping.py

```diff
@@ -446,3 +447,167 @@ async def stress_ping_test(self, port, n):
+        
+        remove_certs(["/tmp/cert.pem", "/tmp/key.pem"])
+
+class WsScheduler(WsPing):
+    
+    """
+    Create 4 connections against 1 backend ws
+    Make 256 async client ws connections, ensure 
+    all ping messages recieved
+    """
+    
+    tempesta = {
+        "config": """
+            listen 81;
+
+            srv_group localhost {
+                server ${server_ip}:8099 conns_n=4;
+            }
+
+            vhost localhost {
+                proxy_pass localhost;
+            }
+
+            http_chain {
+                -> localhost;
+            }
+        """,
+    }
+
+    async def handler(self, websocket, path):
+        global ping_message
+        data = await websocket.recv()
+        reply = f"{data}"
+        print(data)
+        await websocket.send(reply)
+    
+    def test(self):
+        p1 = Process(target=self.run_ws, args=(8099, 1))
+        p2 = Process(target=self.run_test, args=(81, 256))
+        p1.start()
+        self.start_tempesta()
+        p2.start()
+        p2.join()
+        p1.terminate()
+        p2.terminate()
+
+
+class RestartOnUpgrade(WssPing):
+
+    """
+    Test case - we never cache 101 responses
+    First: Send upgrade HTTP connection and - get 101
+    Second: Terminate websocket, call HTTP upgrade again - get 502
+    """
+
+    tempesta = {
+        "config": TEMPESTA_CONFIG % "",
+    }
+
+    async def handler(self, websocket, path):
+        pass
+
+    def call_upgrade(self, port, expected_status):
+        headers_ = {
+            "Host": hostname,
+            "Connection": "Upgrade",
+            "Pragma": "no-cache",
+            "Cache-Control": "no-cache",
+            "User-Agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/99.0.4844.51 Safari/537.36",
+            "Upgrade": "websocket",
+            "Origin": "null",
+            "Sec-WebSocket-Version": "13",
+            "Accept-Encoding": "gzip, deflate",
+            "Sec-WebSocket-Key": "V4wPm2Z/oOIUvp+uaX3CFQ==",
+            "Sec-WebSocket-Accept": "s3pPLMBiTxaQ9kYGzzhZRbK+xOo=",
+            "Sec-WebSocket-Extensions": "permessage-deflate; client_max_window_bits",
+        }
+
+        r = requests.get(f"http://{hostname}:{port}", auth=("user", "pass"), headers=headers_)
+        if r.status_code not in expected_status:
+            self.fail("Test failed cause recieved invalid status_code")
+
+    def fibo(self, n):
+        fib = [0, 1]
+        for i in range(n):
+            fib.append(fib[-2] + fib[-1])
+            if fib[-1] > n:
+                break
+        return fib
+
+    def test(self):
+        p1 = Process(target=self.run_ws, args=(8099,))
+        p1.start()
+        self.start_tempesta()
+        for i in range(1500):
+            if i in self.fibo(1500):
+                self.get_tempesta().restart()
+            self.call_upgrade(81, [101])
+        p1.terminate()
+        remove_certs(["/tmp/cert.pem", "/tmp/key.pem"])
```

Comment:
It's confusing to me. We inherit `RestartOnUpgrade` from `WssPing` but we don't use SSL in this test. 

Some things that looks confusing to me:
1. We create ssl context and certs in `run_ws` but I can't see where context and certs uses. 
2. `WsPing` class have a mixed logic of ssl and not ssl logic, maybe would be better move ssl logic to `WssPing` class? As for me `WsPing` must contain only plain ws logic.
3. Pair gen/remove certs creates for each ssl test. On start and end of the test. It implies we can use decorator for generating certs.

Reply:
Fixed, move TLS/Cert stuff into WSS test